### ISSUE TITLE
Making ArgsTable work with `defineProps` [Vue 3]

### ISSUE THREAD

**Issue by valgeirb** [storybookjs/storybook#16301]
First, thanks for your work on this package.

I'm porting a Vue 2 code base over to Vue 3 using the new `script setup` feature.

I'm using the `defineProps` method to define my props, but after doing this, my props are no longer picked up by the `ArgsTable` component in my `.mdx` stories.

Am I doing something wrong or does `ArgsTable` not support the new `defineProps` feature?

**Additional context**

```vue
<script setup>

const props = defineProps({
  /**
   * A message to display inside the alert
   */
  message: {
    type: String,
    default: "",
  },
  /**
   * The alert has four variants, `error`, `info`, `warning` and `success`
   */
  variant: {
    type: String,
    default: "error",
    validator: (variant) => Object.keys(AlertClasses).includes(variant),
  },
});
</script>
```

**Comment by BombaraGianluca** [comment]
Hi, i red vue-docgen-api now support script setup syntax. 
**Link to documentation**: [https://vue-styleguidist.github.io/docs/Documenting.html#setup-syntax](https://vue-styleguidist.github.io/docs/Documenting.html#setup-syntax)

**The link of github commit:** [https://github.com/vue-styleguidist/vue-styleguidist/releases/tag/v4.44.0](https://github.com/vue-styleguidist/vue-styleguidist/releases/tag/v4.44.0)

You think this feactures will be include in storybook soon? 
Thanks

**Comment by Rigo-m** [comment]
@shilman any news on this? Just kick started a vue 3 project and storybook doesn't seem to pick up types of the props.
It would be a blast if, when writing a story args, I could have type hintings

**Comment by shilman** [comment]
@BombaraGianluca @Rigo-m @valgeirb The latest prerelease of storybook uses the latest release of `vue-docgen-api`, so it might just work out of the box now. To try it out:

```
npx sb upgrade --prerelease
```

Please let me know if it works!

**Comment by shilman** [comment]
@Rigo-m do you have a reproduction you can share? I'd like to take a look and see if I can figure out why it isn't working.

**Comment by Rigo-m** [comment]
Sure! Take a look here: https://github.com/Rigo-m/sb-nuxt-types-repro

**Comment by shilman** [comment]
@Rigo-m I took a quick look and here's what I found.

I added a `console.log({ Title })` to your stories file:

<img width="564" alt="Mouse_Highlight_Overlay" src="https://user-images.githubusercontent.com/488689/153583183-e1003e34-1985-4e26-8a7c-3c47bcee3c33.png">

It looks like `vue-docgen-api` is picking up the `size` prop and not the `title` prop. And it is not  picking up the types for some reason. I ran `vue-docgen-cli` on the file and got the same result, so I think it's an issue with `vue-docgen-api`, or perhaps it's expecting something different in your component definition?

<img width="721" alt="Mouse_Highlight_Overlay" src="https://user-images.githubusercontent.com/488689/153584214-fb03339e-93ef-4059-a65c-f78c36bb81ae.png">

Another thing I found is that you are using the vite builder. Storybook's default builder is webpack, and we use the webpack loader to run `vue-docgen-api`. I guess there must be something similar in the vite builder, but I am not familiar with where that gets set up. However, since the `__docgenInfo` on the component is the same as the output of `vue-docgen-cli`, I assume that's set up correctly and the vite part is a red herring.

**Comment by Rigo-m** [comment]
FYI: using only `defineProps` doesn't fix the situation. Defaulting every prop with `withDefaults` compiler macro exposes every prop (doesn't infer type, but at least docgen reads the props). I guess we can open an issue in `vue-docgen-api` :)

**Comment by Rigo-m** [comment]
@shilman looks like MAYBE we can fix it easily.
It seems we are using vue-docgen-api v4.44.15, but [v.4.44.2](https://github.com/vue-styleguidist/vue-styleguidist/releases/tag/v4.44.2) should fix typescript typings :)

**Comment by BhaskaranR** [comment]
> @shilman looks like MAYBE we can fix it easily. It seems we are using vue-docgen-api v4.44.15, but [v.4.44.2](https://github.com/vue-styleguidist/vue-styleguidist/releases/tag/v4.44.2) should fix typescript typings :)
> 
> EDIT: I guess it's friday and my brain parsed 4.44.2 to be > than 4.44.15. Pardon.

were you able to fix this?

**Comment by Rigo-m** [comment]
@BhaskaranR nope, didn't have any luck

**Comment by avinean** [comment]
@elevatebart Please take a look at this https://github.com/vue-styleguidist/vue-styleguidist/pull/1396
This small change will make life easier.
I'm sure I have not enough context of how things works. But my changes don't break test)
So hope will approve my PR or at least provide better solution.
But please don't ignore it

**Comment by Fsss126** [comment]
Any news on this? Still experiencing this issue with the latest storybook v7.6.10

**Comment by valentinpalkovic** [comment]
@kasperpeulen, @chakAs3 any ideas?

**Comment by shilman** [comment]
Hopefully fixed by #22285

**Comment by chakAs3** [comment]
> @kasperpeulen, @chakAs3 any ideas?

this PR has major fixes and more features for docs #22285.

**Comment by larsrickert** [comment]
Fixed with #22285 in version 8
